src/main/java/org/jenkinsci/plugins/mesos/MesosSlave.java

```diff
@@ -92,6 +111,21 @@ public AbstractCloudComputer createComputer() {
 
   @Override
   protected void _terminate(TaskListener listener) {
-    throw new NotImplementedException();
+    try {
+      logger.info("killing task {}", this.podId);
+      // create a terminating spec for this pod
+      this.getCloud().getMesosClient().killAgent(this.podId);
```

Comment:
Do we have to wait for the kill to succeed?

Reply:
I dont think we do, added a test that verifies kill is successful

Reply:
we may need to do something like Jenkins.getInstance.removeNode(this) so that we would remove this from our record

Reply:
added to my PR, the node removal